### PR TITLE
Create Controllers

### PR DESCRIPTION
**Pull request by Izebiafe**
# Pull Request: Enhance Request Handling in Controllers

## Changes Proposed

This pull request introduces improvements to request handling in controllers. The key changes include:

1. **Controller Usage:** Utilize controllers to handle incoming requests and render empty views. This enhances the organization and separation of concerns in the application.

2. **Rails Naming Conventions:** Adhere to Rails naming conventions to ensure consistency and maintainability across the codebase.

3. **Safe Parameter Usage:** Implement secure practices for utilizing params from browser requests. This ensures that user input is handled safely to prevent security vulnerabilities.

## Implementation Details

- **Controller Handling:**
  - Introduce controllers to manage request processing and view rendering.
  - Organize code to enhance the overall structure of the application.

- **Naming Conventions:**
  - Follow Rails naming conventions for controllers, actions, and views.
  - Ensure that the codebase maintains a consistent and standardized naming pattern.

- **Safe Parameter Usage:**
  - Implement measures to securely use params from browser requests.
  - Mitigate potential security risks associated with handling user input.

## Testing

Comprehensive testing has been conducted to validate the proposed changes. The test suite covers various scenarios to ensure the robustness and reliability of the implemented features.

## Additional Notes

These changes contribute to the overall improvement of the application's architecture and security. Please review the modifications and provide feedback or approval.

Thank you for considering this pull request.

### CODE REVIEW

**Comment by Reem-lab** [comment]
Hello @Izebiafe could you check your slack?

**Review by julie-ify — COMMENTED** [review]
Hi @Izebiafe 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* Kindly correct your linter errors.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: test/controllers/comments_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class CommentsControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (julie-ify) [inline test/controllers/comments_controller_test.rb]: - [ ] You were requested not to use file generation command like scaffold command, but you used it anyways resulting in a lot of unnecessary files. Kindly remove this scaffold file.

File: test/controllers/likes_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class LikesControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (julie-ify) [inline test/controllers/likes_controller_test.rb]: - [ ] You were requested not to use file generation command like scaffold command, but you used it anyways. Kindly remove this scaffold file.

File: test/controllers/posts_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class PostsControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (julie-ify) [inline test/controllers/posts_controller_test.rb]: - [ ] You were requested not to use file generation command like scaffold command, but you used it anyways. Kindly remove this scaffold file.

File: test/controllers/users_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UsersControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (julie-ify) [inline test/controllers/users_controller_test.rb]: - [ ] You were requested not to use file generation command like scaffold command, but you used it anyways. Kindly remove this scaffold file.

File: app/helpers/comments_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module CommentsHelper
+end
```
Comment (julie-ify) [inline app/helpers/comments_helper.rb]: - [ ] You were requested not to use file generation command like scaffold command, but you used it anyways. Kindly remove this scaffold file.

File: app/helpers/likes_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module LikesHelper
```
Comment (julie-ify) [inline app/helpers/likes_helper.rb]: - [ ] You were requested not to use file generation command like scaffold command, but you used it anyways. Kindly remove this scaffold file.

File: app/helpers/posts_helper.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+module PostsHelper
+end
```
Comment (julie-ify) [inline app/helpers/posts_helper.rb]: - [ ] You were requested not to use file generation command like scaffold command, but you used it anyways. Kindly remove this scaffold file.

File: app/helpers/users_helper.rb [hunk 8]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
+end
```
Comment (julie-ify) [inline app/helpers/users_helper.rb]: - [ ] You were requested not to use file generation command like scaffold command, but you used it anyways. Kindly remove this scaffold file.

File: app/controllers/likes_controller.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+class LikesController < ApplicationController
+  def new; end
+
+  def create; end
+end
```
Comment (julie-ify) [inline app/controllers/likes_controller.rb]: - [ ] We have no routes for likes. Kindly remove this controller.

File: app/controllers/comments_controller.rb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+class CommentsController < ApplicationController
+  def new; end
+
+  def create; end
+
+  def destroy; end
+end
```
Comment (julie-ify) [inline app/controllers/comments_controller.rb]: - [ ] Kindly remove this comments controller

File: app/controllers/posts_controller.rb [hunk 11]
```diff
@@ -0,0 +1,11 @@
+class PostsController < ApplicationController
+  def index; end
+
+  def show; end
+
+  def new; end
+
+  def create; end
+
+  def destroy; end
```
Comment (julie-ify) [inline app/controllers/posts_controller.rb]: - [ ] Kindly remove these three controller actions. We only need `index` and `show` actions for the routes we defined.

File: app/views/users/index.html.erb [hunk 12]
```diff
@@ -0,0 +1 @@
+<h1>The Users</h1>
```
Comment (julie-ify) [inline app/views/users/index.html.erb:1]: - [ ] Just as you have the users index page, you should also have the user show page. Kindly create the file.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Izebiafe,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi @Izebiafe, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**